packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ThumbnailCreator.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.imagepicker;
+
+import android.graphics.Bitmap;
+import android.media.ThumbnailUtils;
+import android.os.AsyncTask;
+import android.provider.MediaStore;
+import androidx.core.util.Consumer;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+public class ThumbnailCreator {
+
+  private final ImageResizer imageResizer;
+  private final File externalFilesDirectory;
+
+  public ThumbnailCreator(ImageResizer imageResizer, File externalFilesDirectory) {
+    this.imageResizer = imageResizer;
+    this.externalFilesDirectory = externalFilesDirectory;
+  }
+
+  /** returns the path for the thumbnail image. */
+  public String generateImageThumbnail(String originalImagePath, Double width, Double height) {
+    return imageResizer.resizeImageIfNeeded(originalImagePath, width, height);
+  }
+
+  /** returns the path for the thumbnail image. */
+  public void generateImageThumbnailAsync(
+      final String originalImagePath,
+      final Double width,
+      final Double height,
+      Consumer<String> callback) {
+    BitmapAsyncGenerator generator =
+        new BitmapAsyncGenerator(
+            callback,
+            new Supplier<String>() {
+              @Override
+              public String get() {
+                return generateImageThumbnail(originalImagePath, width, height);
+              }
+            });
+    generator.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
```

Comment:
I think it's safer to just use `execute()` here. We don't need these tasks to run in parallel.

Reply:
in case if need display images in grid layout, without parallel execution, bitmaps will be generated too slow. AsyncTask limit count of parallel task at same time.